scripts/perf-test.sh

```diff
@@ -0,0 +1,191 @@
+#!/bin/bash
+#
+# Copyright Kroxylicious Authors.
+#
+# Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+#
+
+set -eo pipefail
+SCRIPT_DIR=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
+NUM_RECORDS=${NUM_RECORDS:-10000000}
+RECORD_SIZE=${RECORD_SIZE:-1024}
+ON_SHUTDOWN=()
+PERF_TEST=${SCRIPT_DIR}/perf-tests
+GREEN='\033[0;32m'
+YELLOW='\033[1;33m'
+NOCOLOR='\033[0m'
+
+. "${SCRIPT_DIR}/common.sh"
+cd "${SCRIPT_DIR}/.."
+
+runDockerCompose () {
+  docker-compose -f ${PERF_TEST}/docker-compose.yaml "${@}"
+}
+
+doCreateTopic () {
+  local TOPIC
+  ENDPOINT=$1
+  TOPIC=$2
+  docker run --rm --network perf-tests_perf_network quay.io/strimzi/kafka:${STRIMZI_VERSION}-kafka-${KAFKA_VERSION}  \
+      bin/kafka-topics.sh --create --topic ${TOPIC} --bootstrap-server ${ENDPOINT}
+}
+
+doDeleteTopic () {
+  local ENDPOINT
+  local TOPIC
+  ENDPOINT=$1
+  TOPIC=$2
+  docker run --rm --network perf-tests_perf_network quay.io/strimzi/kafka:${STRIMZI_VERSION}-kafka-${KAFKA_VERSION}  \
+      bin/kafka-topics.sh --delete --topic ${TOPIC} --bootstrap-server ${ENDPOINT}
+}
+
+
+warmUp() {
+  producerPerf $1 $2 1
+}
+
+# runs kafka-producer-perf-test.sh transforming the output to an array of objects
+producerPerf() {
+  local ENDPOINT
+  local TOPIC
+  local NUM_RECORDS
+  ENDPOINT=$1
+  TOPIC=$2
+  NUM_RECORDS=$3
+
+  # Input:
+  # 250000 records sent, 41757.140471 records/sec (40.78 MB/sec), 639.48 ms avg latency, 782.00 ms max latency
+  # 250000 records sent, 41757.140471 records/sec (40.78 MB/sec), 639.48 ms avg latency, 782.00 ms max latency, 670 ms 50th, 771 ms 95th, 777 ms 99th, 781 ms 99.9th
+  # Output:
+  # [
+  #  { "sent": 204796, "rate_rps": 40959.2, "rate_mips": 40.00, "avg_lat_ms": 627.9, "max_lat_ms": 759.0 },
+  #  { "sent": 300000, "rate_rps": 43184.108248, "rate_mips": 42.17, "avg_lat_ms": 627.62, "max_lat_ms": 759.00,
+  #    "percentile50": 644, "percentile95": 744, "percentile99": 753, "percentile999": 758 }
+  # ]
+
+  docker run --rm --network perf-tests_perf_network quay.io/strimzi/kafka:${STRIMZI_VERSION}-kafka-${KAFKA_VERSION}  \
+      bin/kafka-producer-perf-test.sh --topic ${TOPIC} --throughput -1 --num-records ${NUM_RECORDS} --record-size ${RECORD_SIZE} \
+      --producer-props acks=all bootstrap.servers=${ENDPOINT} | \
+      jq -R '[.,inputs] | [.[] | match("^(?<sent>\\d+) *records sent" +
```

Comment:
Guys! I'm sorry! `kafka-producer-perf-test.sh -o json`

Reply:
indeed.. the same thoughts were going through my mind.

On the one hand, a enhancement to `kafka-producer-perf-test` and `kafka-consumer-perf-test` to have the option to emit JSON would probably be valued by the community.   On the other hand, I imagine that Kroxylicious's needs will probably outgrow the abilities of KPPT and KCPT.  Looping in @showuon for his take.  There must have been plenty of discussions about making the Kafka CLI json friendly.

The `jq` crap is ugly, but I think it is good enough for now.

Reply:
The transformation from the ouput to a json format was done in this [script](https://github.com/kroxylicious/kroxylicious/blob/main/benchmark_output_to_json.sh) that creates a valid format to be uploaded to kibana. We have to think about this kind of format, because we need to store the data, not only send it to the PR and loose it.

Reply:
If you set env var `KIBANA_OUTPUT_DIR` the script will emit a directory containing producer/consumer json files, one per test.   This hopefully will integrate with what we have already for the Kibana integration.  Obviously we've now got several test cases, where as previously we had one, so there will need to be some downstream changes.